src/DependencyInjection/Configuration.php

```diff
@@ -107,6 +109,39 @@ private function addClientsHosts(NodeBuilder $builder)
             ->defaultValue(27017);
     }
 
+    private function addDriverOptions(NodeBuilder $builder)
+    {
+        $driverOptionsBuilder = $builder
+            ->arrayNode('driverOptions')
+            ->info('Available options for driver')
+            ->children();
+
+        $contextBuilder = $driverOptionsBuilder
+            ->arrayNode('context')
+            ->info('Available options for context')
+            ->children();
+
+        $contextBuilder
+            ->scalarNode('capath')
+            ->defaultNull();
+
+        $contextBuilder
+            ->scalarNode('cafile')
+            ->defaultNull();
+
+        $contextBuilder
+            ->scalarNode('local_cert')
+            ->defaultNull();
+
+        $contextBuilder
+            ->scalarNode('passphrase')
+            ->defaultNull();
+
+        $contextBuilder
+            ->booleanNode('allow_self_signed')
+            ->defaultFalse();
+    }
```

Comment:
This level of granularity in the configuration is error prone and is very boring to maintain. This is why I whould prefer to delegate completely the driver options building to a service of the consumer app.